### ISSUE TITLE
May the Gan hinge loss be negative?

### ISSUE THREAD

**Issue by wvinzh** [NVlabs/SPADE#122]
https://github.com/NVlabs/SPADE/blob/1a687baeada266a3c92be41295f1ea3d5efd4f93/models/networks/loss.py#L75
As the code shows, in  `-torch.mean(input)` , the `input` is produced by a convolution layer in the discriminator, so the loss might be negative sometimes, but the VGG loss and GAN_Feat loss are always positive. The final loss is the sum of these losses, so how does the negative value influence the gradient?

**Comment by Alpe6825** [comment]
> https://github.com/NVlabs/SPADE/blob/1a687baeada266a3c92be41295f1ea3d5efd4f93/models/networks/loss.py#L75
> 
> 
> As the code shows, in `-torch.mean(input)` , the `input` is produced by a convolution layer in the discriminator, so the loss might be negative sometimes, but the VGG loss and GAN_Feat loss are always positive. The final loss is the sum of these losses, so how does the negative value influence the gradient?

I am also interested in this question.
